azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/ConfigMojo.java

```diff
@@ -0,0 +1,319 @@
+import com.microsoft.azure.management.resources.fluentcore.arm.Region;
+import com.microsoft.azure.maven.appservice.PricingTierEnum;
+import com.microsoft.azure.maven.queryer.MavenPluginQueryer;
+import com.microsoft.azure.maven.queryer.QueryFactory;
+import com.microsoft.azure.maven.webapp.configuration.Deployment;
+import com.microsoft.azure.maven.webapp.configuration.DeploymentSlotSetting;
+import com.microsoft.azure.maven.webapp.configuration.OperatingSystemEnum;
+import com.microsoft.azure.maven.webapp.configuration.RuntimeSetting;
+import com.microsoft.azure.maven.webapp.configuration.SchemaVersion;
+import com.microsoft.azure.maven.webapp.handlers.WebAppPomHandler;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.codehaus.plexus.util.StringUtils;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+@Mojo(name = "config", defaultPhase = LifecyclePhase.VALIDATE)
+public class ConfigMojo extends AbstractWebAppMojo {
+
+    public static final String NOT_EMPTY_REGEX = "[\\s\\S]+";
+    public static final String BOOLEAN_REGEX = "[YyNn]";
+
+    public static final String CONFIGURATION_NOT_FOUND = "Configuration not found, init new configuration.";
+    public static final String CONVERT_TO_V2SCHEMA = "Converting configuration to V2 schema";
+    public static final String SAVING_TO_POM = "Saving configuration to pom.";
+    public static final String EXITING = "Config only support V2 schema, exiting";
+
+    private MavenPluginQueryer queryer;
+    private WebAppPomHandler pomHandler;
+    private static final String[] configTypes = {"Application", "Runtime", "DeploymentSlot"};
+
+    @Override
+    protected void doExecute() throws Exception {
+        queryer = QueryFactory.getQueryer(settings, getLog());
+        pomHandler = new WebAppPomHandler("pom.xml");
+
+        try {
+            final WebAppConfiguration configuration = getWebAppConfigurationFromPom();
+            if (configuration != null &&
+                SchemaVersion.fromString(configuration.getSchemaVersion()) == SchemaVersion.V1) {
+                convertToV2Schema(configuration);
+            } else {
+                config(configuration);
+            }
+        } finally {
+            queryer.close();
+        }
+    }
+
+    protected void convertToV2Schema(WebAppConfiguration configuration) throws IOException, MojoFailureException {
+        final String result = queryer.assureInputFromUser("convertToV2", "Y", BOOLEAN_REGEX, "Convert to V2 Schema?" +
+            "(Y/N) : ", null);
+        if (result.toLowerCase().equals("y")) {
+            info(CONVERT_TO_V2SCHEMA);
+            pomHandler.convertToV2Configuration(configuration);
```

Comment:
Do we have different configurations for V1 & V2?

Reply:
no, i will rename it to convertToV2Schema